charmcraft/templates/init-kubernetes/tox.ini.j2

```diff
@@ -49,6 +49,7 @@ description = Run unit tests
 deps =
     pytest
     coverage[toml]
+    ops[testing]
```

Comment:
I wonder if a version specifier should be used here.
After all the testing extra has only been introduced in 2.17.0

Reply:
I wondered about that too. If you somehow have a requirement of ops < 2.17, you do get a warning:

```
WARNING: ops 2.16.1 does not provide the extra 'testing'
```

And obviously the tests will fail.

I think it's not necessary because the main project dependencies specify `ops ~= 2.17`, and that's included via the `-r requirements.txt` below. If someone for some reason changed the requirements file to be < 2.17 then things will break but also they need to rewrite all their tests, so I think they need to take responsibility for the cleanup.

Reply:
I'd add `>= 2.17.0` to make this explicit.

If someone somehow has a `< xx` specifier somewhere else, it's better for version resolution to fail than for unit tests for fail, isn't it?